docs/source/_notebooks/xgboost-titanic.rst

```diff
@@ -0,0 +1,2468 @@
+
+Explaining XGBoost predictions on the Titanic dataset
+=====================================================
+
+This tutorial will show you how to analyze predictions of a tree
+ensemble classifier (XGBoost in this case, but it also works for tree
+ensembles from scikit-learn and for regression). We will use `Titanic
+dataset <https://www.kaggle.com/c/titanic/data>`__, which is small and
+has not too many features, but is still rich enough.
+
+1. Training data
+----------------
+
+Let's start by loading the data:
+
+.. code:: python
+
+    import csv
+    import numpy as np
+    
+    with open('titanic-train.csv', 'rt') as f:
+        data = list(csv.reader(f))
+
+Variable descriptions:
+
+-  **Survival:** Survival (0 = No; 1 = Yes)
+-  **Pclass:** Passenger Class (1 = 1st; 2 = 2nd; 3 = 3rd)
+-  **Name:** Name
+-  **Sex:** Sex
+-  **Age:** Age
+-  **Sibsp:** Number of Siblings/Spouses Aboard
+-  **Parch:** Number of Parents/Children Aboard
+-  **Ticket:** Ticket Number
+-  **Fare:** Passenger Fare
+-  **Cabin:** Cabin
+-  **Embarked:** Port of Embarkation (C = Cherbourg; Q = Queenstown; S =
+   Southampton)
+
+Next, shuffle data and separate features from what we are trying to
+predict: survival.
+
+.. code:: python
+
+    from pprint import pprint
+    from sklearn.utils import shuffle
+    from sklearn.model_selection import train_test_split
+    
+    feature_names = data[0][2:]
+    _all_xs = [dict(zip(feature_names, row[2:])) for row in data[1:]]
+    _all_ys = np.array([int(row[1]) for row in data[1:]])
+    
+    all_xs, all_ys = shuffle(_all_xs, _all_ys, random_state=0)
+    train_xs, valid_xs, train_ys, valid_ys = train_test_split(
+        all_xs, all_ys, test_size=0.25, random_state=0)
+    print('{} items total, {:.1%} true'.format(len(all_xs), np.mean(all_ys)))
+    pprint(valid_xs[:2])
+
+
+.. parsed-literal::
+
+    891 items total, 38.4% true
+    [{'Age': '19',
+      'Cabin': '',
+      'Embarked': 'S',
+      'Fare': '10.1708',
+      'Name': 'Dakic, Mr. Branko',
+      'Parch': '0',
```

Comment:
I think it's not a problem for classification here: 0 will be treated as "special" value by the decision tree (but this seems ok), and there are no really missing values in "Parch", and zero here means that there are no parents/children abroad.
For visualising predictions this is more problematic, because we filter out missing values.
So in general this is bad, let me think how it can be solved...